vault/resource_pki_secret_backend_config_scep.go

```diff
@@ -0,0 +1,211 @@
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cert": {
+							Type:        schema.TypeMap,
+							Optional:    true,
+							Description: "The accessor and cert_role properties for cert auth backends",
+						},
+						"scep": {
+							Type:        schema.TypeMap,
+							Optional:    true,
+							Description: "The accessor property for SCEP auth backends",
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+			consts.FieldExternalValidation: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Description: "Lists the 3rd party validation of SCEP requests",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"intune": {
+							Type:        schema.TypeMap,
+							Optional:    true,
+							Description: "The credentials to enable Microsoft Intune validation of SCEP requests",
+							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+								isClientSecret := strings.HasSuffix(k, "client_secret")
+								return isClientSecret
+							},
+						},
+					},
+				},
+			},
+			consts.FieldLastUpdated: {
+				Type:        schema.TypeString,
+				Computed:    true, // read-only property
+				Description: "A read-only timestamp representing the last time the configuration was updated",
+			},
+		},
+	}
+}
+
+func pkiSecretBackendConfigScepWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if err := verifyPkiScepFeatureSupported(meta); err != nil {
+		return diag.FromErr(err)
+	}
+
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	backend := d.Get(consts.FieldBackend).(string)
+	path := pkiSecretBackendConfigScepPath(backend)
+
+	fieldsToSet := []string{
+		consts.FieldEnabled,
+		consts.FieldDefaultPathPolicy,
+		consts.FieldAllowedEncryptionAlgorithms,
+		consts.FieldAllowedDigestAlgorithms,
+		consts.FieldRestrictCAChainToIssuer,
+	}
+
+	data := map[string]interface{}{}
+	for _, field := range fieldsToSet {
+		if val, ok := d.GetOk(field); ok {
+			data[field] = val
+		}
+	}
+
+	if authenticators, ok := getListOfNotEmptyMaps(d, consts.FieldAuthenticators); ok {
+		data[consts.FieldAuthenticators] = authenticators
+	}
+
+	if validations, ok := getListOfNotEmptyMaps(d, consts.FieldExternalValidation); ok {
+		data[consts.FieldExternalValidation] = validations
+	}
+
+	log.Printf("[DEBUG] Updating SCEP config on PKI secret backend %q:\n%v", backend, data)
+	_, err := client.Logical().WriteWithContext(ctx, path, data)
+	if err != nil {
+		return diag.Errorf("error updating SCEP config for PKI secret backend %q: %s", backend, err)
+	}
+	log.Printf("[DEBUG] Updated SCEP config on PKI secret backend %q", backend)
+
+	d.SetId(path)
+
+	return pkiSecretBackendConfigScepRead(ctx, d, meta)
+}
+
+func getListOfNotEmptyMaps(d *schema.ResourceData, field string) (any, bool) {
+	if raw, ok := d.GetOk(field); !ok {
+		return nil, false
+	} else {
```

Comment:
No need for the else block here.

Reply:
I do, since `raw` is used in the other block, but I pulled the variables outside the if statement since it is a small function anyway.